engines/pegasus/pegasus.cpp

```diff
@@ -702,9 +702,18 @@ void PegasusEngine::writeContinueStream(Common::WriteStream *stream) {
 }
 
 Common::StringArray PegasusEngine::listSaveFiles() {
+	const Common::String autoSaveName("pegasus-AutoSave.sav");
 	Common::StringArray fileNames = g_system->getSavefileManager()->listSavefiles("pegasus-*.sav");
+	// Autosave must be at slot 0, so remove it, then prepend (even if it doesn't exist,
+	// it will be prepended)
+	Common::StringArray::iterator it = Common::find(fileNames.begin(), fileNames.end(), autoSaveName);
+	if (it != fileNames.end())
+		fileNames.erase(it);
 	Common::sort(fileNames.begin(), fileNames.end());
-	return fileNames;
+	Common::StringArray res;
+	res.push_back(autoSaveName);
+	Common::copy(fileNames.begin(), fileNames.end(), res.end());
```

Comment:
I think `Common::copy` requires the destination to be already allocated? Which means here it would copy the `fileNames` element to unallocated memory, causing random issues.

You can use `res.push_back(fileNames);`, but I think the easiest solution is to insert the autosave name at the start of `fileNames` and not use another array (i.e. either use `fileNames.insert(fileNames.begin(), autoSaveName)`, or `fileNames.insert_at(0, autoSaveName)`.

Reply:
I am thinking we might need to also modify the implementation of `PegasusEngine::saveGameState()` so that if `isAutosave` is true it uses the `"pegasus-AutoSave.sav"` name instead of generating one from the save description (which might be `AutoSave` by default, but I am wondering if it might get translated for example, so not using the description for the AutoSave game and hardcoding the name there as well seems safer).

Reply:
I was looking for `push_front`. Didn't consider `insert_at`, thanks for that :)